template/cordova/lib/deployment.js

```diff
@@ -0,0 +1,272 @@
+ * @shouldLaunch Indicates whether to launch the app after installing it.
+ * @shouldUpdate Indicates whether to explicitly update the app, or install clean.
+ * @pin Optionally provided if the device requires pairing for deployment.
+ */
+DeploymentTool.prototype.installAppPackage = function(pathToAppxPackage, targetDevice, shouldLaunch, shouldUpdate, pin) {
+    return Q.reject('May not use DeploymentTool directly, instead get an instance from DeploymentTool.getDeploymentTool()');
+};
+
+/**
+ * Uninstalls an app package from the target device.
+ * @returns A Promise which will be fulfilled on success or rejected on failure.
+ * @param packageInfo The app package name or Phone GUID representing the app.
+ * @param targetDevice An object returned from a successful call to enumerateDevices.
+ */
+DeploymentTool.prototype.uninstallAppPackage = function(packageInfo, targetDevice) {
+    return Q.reject('Unable to uninstall any app packages because that feature is not supported.');
+};
+
+/**
+ * Gets a list of installed apps on the target device.  This function is not supported for 
+ * Windows Phone 8.1.
+ * @param targetDevice {Object} An object returned from a successful call to enumerateDevices.
+ * @returns A Promise for an array of app names.
+ */
+DeploymentTool.prototype.getInstalledApps = function(targetDevice) {
+    return Q.reject('Unable to get installed apps because that feature is not supported.');
+};
+
+/**
+ * Launches an app on the target device.  This function is not supported for Windows 10.
+ * @param packageInfo {String} The app package name or Phone GUID representing the app.
+ * @param targetDevice {Object} An object returned from a successful call to enumerateDevices.
+ * @returns A Promise for when the app is launched.
+ */
+DeploymentTool.prototype.launchApp = function(packageInfo, targetDevice) {
+    return Q.reject('Unable to launch an app because that feature is not supported.');
+};
+
+/**
+ * Gets a DeploymentTool for use in 
+ */
+DeploymentTool.getDeploymentTool = function(targetOsVersion) {
+    if (targetOsVersion === '8.0' || targetOsVersion === '8.1') {
+        return new AppDeployCmdTool(targetOsVersion);
+    }
+    else {
+        return new WinAppDeployCmdTool(targetOsVersion);
+    }
+};
+
+// DeviceInfo is an opaque object passed to install/uninstall.
+// Implementations of DeploymentTool can populate it with any additional
+//  information required for accessing them.
+function DeviceInfo(deviceIndex, deviceName, deviceType) {
+    this.index = deviceIndex;
+    this.name = deviceName;
+    this.type = deviceType;
+}
+
+DeviceInfo.prototype.toString = function() {
+    return this.index + '. ' + this.name + ' (' + this.type + ')';
+};
+
+function AppDeployCmdTool(targetOsVersion) {
+    if (!(this instanceof AppDeployCmdTool))
+        throw new ReferenceError('Only create an AppDeployCmdTool as an instance object.');
+
+    DeploymentTool.call(this);
```

Comment:
No, but it's convention because AppDeployCmdTool "derives from" DeploymentTool.  The fact that DeploymentTool doesn't do anything in its constructor would make eliminating this a micro-optimization, but is unnecessary.